File: leetcode-submissions/929_Unique_Email_Addresses.cpp
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        unordered_set<string>st;
        for(string& email : emails){
            size_t atPos = email.find('@');
            string local = email.substr(0 , atPos);
            string domain = email.substr(atPos+1);
            string modified_local = "";
            for(char& ch : local){
                if(ch=='.'){
                    continue;
                }else if(ch=='+'){
                    break;
                }
                
               modified_local+=ch;
            }
            modified_local+='@';
            modified_local+=domain;
            st.insert(modified_local);
        }
        return st.size();
    }
};

File: leetcode-submissions/929_Unique_Email_Addresses.ts
function numUniqueEmails(emails: string[]): number {
    //story points 
    // 1 . remove the part after + and remove all the dots 
    // 2. store all the clean emails into the set
    // 3. return the size of the set 
    const st:Set<string>= new Set();
    for(const email of emails){
        const [local , domain]= email.split('@');
        let cleanedLocal :string = "";
        for(let i:number =0 ;i<local.length;i++){
            if(local[i]=='.' ){
                continue ;
            }
            if(local[i]==='+') break;
            cleanedLocal+=local[i];
        }
        const cleanedEmail = `${cleanedLocal}@${domain}`
        st.add(cleanedEmail);
    }
    return st.size;
    
};